File: WebAPI/FMS.Service/DTOs/CreditCardStatement.cs
using FMS.Entity.Context.Models;
using System;
using System.Collections.Generic;
using System.Text;

namespace FMS.Service.DTOs
{
    public class CreditCardStatement
    {
        // CreditCard
        public int CreditCardId { get; set; }
        public string CreditCardNumber { get; set; }
        public string CreditCardName { get; set; }
        public PayeeType PayeeType { get; set; }
        public decimal Balance { get; set; }

        // Transactions
        public List<Transaction> Transactions { get; set; }

        public class Transaction
        {
            // Output
            // cc output transactions
            // from cc transactions
            // cc is paying off $ to payee
            public int CreditCardTransactionId { get; set; }
            public int PayeeId { get; set; }
            public string PayeeName { get; set; }
            public PayeeType PayeeType { get; set; }
            public decimal AmountPaid { get; set; }
            public DateTime TransactionDate { get; set; }
            public TransactionStatus TransactionStatus { get; set; }
            public decimal CurrentBalance { get; set; }
            public decimal LastBalance { get; set; }

            // Input
            // cc input transactions
            // from bank transactions
            // cc is getting $ from bank
            public int BankId { get; set; }
            public string BankName { get; set; }
            public int AccountId { get; set; }
            public int AccountNumber { get; set; }

            // decides either In or Out
            public TransactionType TransactionType { get; set; }

            public string RefCode { get; set; }

        }
    }

   
}

File: Angular/src/app/models/creditCardStatement.ts

export default class CreditCardStatement {
    creditCardId: number;
    creditCardNumber: string;
    creditCardName : string;
    payeeType: number;
    payeeTypeDisplay: string;
    balance: number;
    transactions: Transaction[];

    groupBy: string;
    // '',transactionDate, payeeName

    ccColor : string;

    // summary report
    totalIn: number;
    totalOut: number;
    payeeTypeSummaryDisplay: string;
    startDate: Date;
    endDate: Date;
}

class Transaction {
   
    creditCardTransactionId: number;
    payeeId: number;
    payeeName: string;
    payeeType: number;
    payeeTypeDisplay: string;
    amountPaid: number;
    transactionDate: Date;
    transactionStatus: number;
    lastBalance: number;
    currentBalance: number;

    transactionDateString: string;

    bankId : number;
    bankName : string;
    accountId: number;
    accountNumber : number;

    // decides $In or $Out
    transactionType : number;
    transactionTypeDisplay : string;

    refCode : string;
}